cdap-data-fabric/src/main/java/io/cdap/cdap/data2/metadata/lineage/field/FieldLineageInfo.java

```diff
@@ -162,6 +170,8 @@ private void computeAndValidateFieldLineageInfo(Collection<? extends Operation>
                     "operation '%s'.", read.getName()));
           }
           readOperations.add(read);
+          for (String output: read.getOutputs())
```

Comment:
We don't need this, this will also not work for the following cases: Read: [body] -> Transform: [body] -> [input], Transform [input] -> [], this will not find the correct origin for input

Reply:
The correct origin for input in that case (which was a rename from body) would be Read, right? I have a unit tests that verifies this case, and it would indeed return Read as the origin.